xm-commons-permission/src/main/java/com/icthh/xm/commons/permission/service/PermissionCheckService.java

```diff
@@ -146,124 +156,210 @@ private boolean checkPermission(Authentication authentication,
         resources.put("oauth2", new OAuth2SecurityExpressionMethods(authentication));
 
         Map<String, String> env = new HashMap<>();
-        env.put(EnvironmentVariable.IP.getName(), xmAuthenticationContextHolder
-            .getContext().getRemoteAddress().orElse(null));
-        resources.put("env", env);//put some env variables in this map
+        env.put(
+            EnvironmentVariable.IP.getName(),
+            xmAuthenticationContextHolder.getContext().getRemoteAddress().orElse(null)
+        );
+        resources.put("env", env);
 
         StandardEvaluationContext context = new StandardEvaluationContext();
         context.setVariables(resources);
         context.registerFunction("getRequestHeader", GET_REQUEST_HEADER);
 
-        Permission permission = getPermission(roleKey, privilegeKey);
+        Collection<Permission> permissions = getPermissions(roleKey, privilegeKey);
 
-        if (!isPermissionValid(permission)) {
-            log(logPermission, Level.ERROR,
+        if (!isPermissionEnabled(permissions)) {
+            log(logPermission,
+                Level.ERROR,
                 "access denied: privilege={}, role={}, userKey={} due to privilege is not permitted",
-                privilegeKey, roleKey, getUserKey());
+                privilegeKey,
+                roleKey,
+                getUserKey()
+            );
             return false;
         }
+
         boolean validCondition = true;
-        if (!isConditionValid(permission.getEnvCondition(), context)) {
-            log(logPermission, Level.ERROR,
-                "access denied: privilege={}, role={}, userKey={} due to env condition: [{}] with context [{}]",
-                privilegeKey, roleKey, getUserKey(), permission.getEnvCondition().getExpressionString(), resources);
+
+        if (isEnvConditionNotValid(permissions, context)) {
+            log(logPermission,
+                Level.ERROR,
+                "access denied: privilege={}, role={}, userKey={} due to env condition: {} with context [{}]",
+                privilegeKey,
+                roleKey,
+                getUserKey(),
+                permissions.stream()
+                    .map(Permission::getEnvCondition)
+                    .filter(Objects::nonNull)
+                    .map(Expression::getExpressionString)
+                    .collect(toList()),
+                resources
+            );
             validCondition = false;
         }
-        if (checkCondition && !isConditionValid(permission.getResourceCondition(), context)) {
-            log(logPermission, Level.ERROR,
-                "access denied: privilege={}, role={}, userKey={} due to env condition: [{}] with context [{}] "
+
+        if (checkCondition && isResourceConditionNotValid(permissions, context)) {
+            log(logPermission,
+                Level.ERROR,
+                "access denied: privilege={}, role={}, userKey={} due to env condition: {} with context [{}] "
```

Comment:
not ur fault but would be nice to fix: this log duplicates above one with redundant "with context [{}]" string